app/models/alert.rb

```diff
@@ -1,15 +1,44 @@
 class Alert < ApplicationRecord
+  ALERT_PERMIT = %i(
+    message origin status via_email owner_id owner_type zone_id
+  ).freeze
+
+  enum status: {pending: 0, resolved: 1, ignored: 2}
+  enum origin: {sensor_threshold: 0, sensor_error: 1, ml_detection: 2,
+                manual_input: 3}
+
   has_one_attached :snapshot
+  belongs_to :owner, polymorphic: true
+  belongs_to :zone
+  belongs_to :user, optional: true
+
+  validates :message, presence: true
+  validates :owner, presence: true
+  validates :zone, presence: true
 
-  enum status: {PENDING: 0, RESOLVED: 1, IGNORED: 2}
-  enum origin: {
-    SENSOR_THRESHOLD: 0,
-    SENSOR_ERROR: 1,
-    ML_DETECTION: 2,
-    MANUAL_INPUT: 3
+  scope :newest, ->{order(created_at: :desc)}
+  scope :with_status, lambda {|status|
+                        if status.present? && statuses.key?(status)
+                          where(status:)
+                        end
+                      }
```

Comment:
```suggestion
  scope :with_status, (lambda do |status|
    if status.present? && statuses.key?(status)
      where(status:)
    end
  end)
```
lambda thì e dùng do .. end nha, fix tương tự cho những chỗ khác